actioncable/app/javascript/action_cable/subscriptions.js

```diff
@@ -29,10 +29,15 @@ export default class Subscriptions {
   // Private
 
   add(subscription) {
+    const existing = this.findAll(subscription.identifier).length > 0
     this.subscriptions.push(subscription)
     this.consumer.ensureActiveConnection()
     this.notify(subscription, "initialized")
-    this.subscribe(subscription)
+    if (!existing) {
```

Comment:
I'd extract `this.findAll(subscription.identifier).length > 0` to a method (e.g: `contains(identitfier)`) and also remove the separated local variable. You could do:

```js
if (!contains(identifier)) {
   ...
}